### PR TITLE
510 add cicd for new socket service

### PR DESCRIPTION
**Pull request by thanhphanvanncc**
# Ticket
PR: #510 

# General
- [x] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [x] I fully understand the code I submitted.
- [x] No hardcoded configuration or secrets.
- [x] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: .github/workflows/deploy-socket-service.yml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+          args: --timeout=5m --output.text.print-issued-lines
+        continue-on-error: true
+
+      - name: Build WebSocket Service binary
+        run: |
+          mkdir -p build 
+          cd socket-service
+          GOOS=linux GOARCH=amd64 go build -o ../build/socket-service .
+
+      - name: Get short SHA
```
Comment (tuananhncc) [inline .github/workflows/deploy-socket-service.yml]: Maybe use the release tag instead of git sha. Need confirmation.
